thunder/__init__.py

```diff
@@ -692,6 +716,7 @@ def fn_(*args, **kwargs) -> Any:
 
     if isinstance(fn, pytorch.nn.Module):
         fn_ = ThunderModule(fn, fn_)
+        cd._thunder_module_map[id(fn)] = fn_
```

Comment:
What about the case that you jit a function that takes the module? We won't have this reference so easily

Reply:
Yeah, but it needs extension to be of full use, but it is not a regression as is. I added this as a map because it enables the case where we do have modules. We still need to add the mechanism to populate this map.